altapay.php

```diff
@@ -0,0 +1,2353 @@
+                ->setModuleName($this->name)
+                ->setAdditionalInformation($template)
+                ->setLogo(Media::getMediaPath(_PS_MODULE_DIR_ . $this->name . '/views/img/payment_icons/'
+                    . $paymentMethod['icon_filename']));
+            $paymentsOptions[] = $paymentOptions;
+        }
+
+        echo '<script src="https://cdn.jsdelivr.net/npm/js-cookie@beta/dist/js.cookie.min.js"></script>';
+        return $paymentsOptions;
+    }
+
+
+    /**
+     * Hook for binding custom JS files on prestashop front end
+     * @param $params
+     */
+    public function hookActionFrontControllerSetMedia($params)
+    {
+        $this->context->controller->addJquery();
+        $this->context->controller->addJS($this->_path . '/views/js/creditCardFront.js', 'all');
+    }
+
+    /**
+     * Method to get template variable information like path, ssl path, methods
+     * @return array
+     */
+    public function getTemplateVarInfos()
+    {
+        $cart = $this->context->cart;
+        $currency = $this->getCurrencyForCart($cart);
+        $paymentMethods = Terminal::getActiveTerminalsForCurrency($currency->iso_code);
+
+        return array(
+            'this_path' => $this->_path,
+            'this_path_altapay' => $this->_path,
+            'this_path_ssl' => Tools::getShopDomainSsl(true, true) . __PS_BASE_URI__ . 'modules/' . $this->name . '/',
+            'methods' => $paymentMethods,
+            'PS_STOCK_MANAGEMENT' => Configuration::get('PS_STOCK_MANAGEMENT'),
+        );
+    }
+
+    /**
+     * Hook triggered at the time of payment returns
+     * @param $params
+     */
+    public function hookPaymentReturn($params)
+    {
+        if (!$this->active) {
+            return;
+        }
+        //Prestashop 1.7 doesn't have the $params['objOrder']
+        if (!isset($params['objOrder']) || !is_object($params['objOrder'])) {
+            $params['objOrder'] = $params['order'];
+        }
+
+        $state = $params['objOrder']->getCurrentState();
+        $results = Db::getInstance()->getRow('SELECT * 
+        FROM `' . _DB_PREFIX_ . 'altapay_order` WHERE id_order=' . $params['objOrder']->id);
+        if ($state == Configuration::get('PS_OS_PAYMENT') || $state == Configuration::get('PS_OS_OUTOFSTOCK')) {
+            $this->smarty->assign(array(
+                'total_to_pay' => Tools::displayPrice($params['total_to_pay'], $params['currencyObj'], false),
+                'status' => 'ok',
+                'unique_id' => $results['unique_id'],
+                'payment_id' => $results['payment_id'],
+                'id_order' => $params['objOrder']->id
+            ));
+            if (isset($params['objOrder']->reference) && !empty($params['objOrder']->reference)) {
+                $this->smarty->assign('reference', $params['objOrder']->reference);
+            }
+        } else {
+            //if 'open' = Configuration::get('ALTAPAY_OS_PENDING')
```

Comment:
Maybe it would be best to apply this check and throw an exception if it's not the case.